src/sandstorm/grain.capnp

```diff
@@ -739,7 +752,17 @@ struct RoleDef {
   # "default", then such sharing actions will be treated as having an empty permissions set (the
   # user can open the grain, but the grain is told that the user has no permissions).
   #
+  # This field should *not* be used to specify a default role hint for sharing!  For that, use
+  # `defaultSelection` immediately below.
+  #
   # See also `ViewSharingLink.RoleAssignment.none`, below.
+
+  defaultSelection @6 :Bool = false;
```

Comment:
I thought the point of having a separate field was so that we could unbind the two.

Actually, hmm.  Now I find myself wishing that `default` and `defaultSelection` were integers on `ViewInfo`, rather than booleans on `RoleDef`, given that it only makes sense for one role to be default for upgrading the old permissions model, and for only one role to be default in selection UI.

I think I may instead back this piece of the change out for now, and maybe do it as a separate piece later.